webapp/src/modules/contract/utils.spec.ts

```diff
@@ -0,0 +1,249 @@
+        expect(upsertContracts(storedContracts, newContracts)).toEqual([
+          ...storedContracts,
+          ...newContracts
+        ])
+      })
+    })
+
+    describe('and newContracts has a contract with same address but different chain id', () => {
+      beforeEach(() => {
+        newContracts = [
+          {
+            address: 'other address',
+            chainId: ChainId.ETHEREUM_GOERLI,
+            name: 'name'
+          } as Contract
+        ]
+      })
+
+      it('should return an array with the new contract added to the stored contracts', () => {
+        expect(upsertContracts(storedContracts, newContracts)).toEqual([
+          ...storedContracts,
+          ...newContracts
+        ])
+      })
+    })
+
+    describe('and newContracts has a contract with the same address and chain id but different name', () => {
+      beforeEach(() => {
+        newContracts = [
+          {
+            address: 'ADDRESS',
+            chainId: ChainId.MATIC_MUMBAI,
+            name: 'different name'
+          } as Contract
+        ]
+      })
+
+      it('should update the stored contract with the new name', () => {
+        expect(upsertContracts(storedContracts, newContracts)).toEqual([
+          { ...storedContracts[0], name: newContracts[0].name }
+        ])
+      })
+    })
+  })
+})
+
+describe('when calling getContractKey', () => {
+  it('should return a string with the contract address and chain id', () => {
+    expect(
+      getContractKey({
+        address: 'address',
+        chainId: ChainId.MATIC_MUMBAI
+      } as Contract)
+    ).toEqual('address-80001')
+  })
+})
+
+describe('when calling getContractKeyFromNFT', () => {
+  it('should return a string with the contract address and chain id', () => {
+    expect(
+      getContractKeyFromNFT({
+        contractAddress: 'address',
+        chainId: ChainId.MATIC_MUMBAI
+      } as NFT)
+    ).toEqual('address-80001')
+  })
+})
+
+describe('when calling getAuthorizationKey', () => {
+  it('should return a string with the contract address and chain id', () => {
+    expect(
+      getAuthorizationKey({
+        address: 'address',
+        authorizedAddress: 'authorizedAddress',
+        contractAddress: 'contractAddress',
+        chainId: ChainId.MATIC_MUMBAI
+      } as Authorization)
+    ).toEqual('address-authorizedAddress-contractAddress-80001')
+  })
+})
+
+describe('when calling getStubMaticCollectionContract', () => {
+  describe('and the provided address is upper case', () => {
+    it('should return stub matic collection contract with the address lowercased', () => {
+      expect(getStubMaticCollectionContract('ADDRESS')).toEqual({
+        address: 'address',
+        category: NFTCategory.WEARABLE,
+        chainId: undefined,
+        name: STUB_MATIC_COLLECTION_CONTRACT_NAME,
+        network: Network.MATIC,
+        vendor: VendorName.DECENTRALAND
+      })
+    })
+  })
+})
```

Comment:
As it's only one case, you can remove the describe that is used for the context if you want and just have the address be lowercased:
```suggestion
describe('when calling getStubMaticCollectionContract', () => {
    it('should return stub matic collection contract with the address lowercased', () => {
      expect(getStubMaticCollectionContract('ADDRESS')).toEqual({
        address: 'address',
        category: NFTCategory.WEARABLE,
        chainId: undefined,
        name: STUB_MATIC_COLLECTION_CONTRACT_NAME,
        network: Network.MATIC,
        vendor: VendorName.DECENTRALAND
      })
    })
})
```
Also, why is the `chainId` undefined?